doc/syntax.md

```diff
@@ -0,0 +1,476 @@
+
+|Quantity | Unit Name | NESTML/SI unit |
+|---|---|---|
+| length              | metre    | m |
+| mass                | kilogram | kg |
+| time                | second   | s |
+| electric current    | ampere   | A |
+| temperature         | kelvin   | K |
+| amount of substance | mole     | mol |
+| luminous intensity  | candela  | cd |
+
+Any other physical unit can be expressed as a combination of these seven units. These other units are called derived units. NESTML provides the following derived units as built-in data types:
+
+|Quantity | Unit name | Definition | NESTML/SI unit |
+|---|---|---|---|
+| capacitance            | farad   | A<sup>2</sup> s<sup>4</sup> kg<sup>−1</sup> m<sup>−2</sup> | F |
+| electrical conductance | siemens | A<sup>2</sup> s<sup>3</sup> kg<sup>−1</sup> m<sup>−2</sup> | S |
+| electric charge        | coulomb | A s                                                        | C |
+| electric potential     | volt    | kg m<sup>2</sup> A<sup>−1</sup> s<sup>−3</sup>             | V |
+| electric resistance    | ohm     | kg m<sup>2</sup> A<sup>−2</sup> s<sup>−3</sup>             | ohm/&‌#x03A9; |
+| frequency              | hertz   | s<sup>−1</sup>                                             | Hz |
+
+| | | | |
+|---|---|---|---|
+|||| Bq |
+|||| Gy |
+|||| Sv |
+|||| J |
+|||| W  |
+|||| Wb |
+|||| N |
+|||| Pa |
+|||| kat |
+
+Units can have at most one of the following magnitude prefixes:
+
+|Factor | SI Name | NESTML prefix | Factor | SI Name | NESTML prefix
+|---|---|---|---|---|---|
+|10<sup>-1</sup>  | deci  | d  | 10<sup>1</sup>  | deca  | da |
+|10<sup>-2</sup>  | centi | c  | 10<sup>2</sup>  | hecto | h |
+|10<sup>-3</sup>  | milli | m  | 10<sup>3</sup>  | kilo  | k |
+|10<sup>-6</sup>  | micro | mu | 10<sup>6</sup>  | mega  | M |
+|10<sup>-9</sup>  | nano  | n  | 10<sup>9</sup>  | giga  | G |
+|10<sup>-12</sup> | pico  | p  | 10<sup>12</sup> | tera  | T |
+|10<sup>-15</sup> | femto | f  | 10<sup>15</sup> | peta  | P |
+
+| | | | | | |
+|---|---|---|---|---|---|
+|10<sup>-18</sup> | atto  | a  | 10<sup>18</sup> | exa   | E |
+|10<sup>-21</sup> | zepto | z  | 10<sup>21</sup> | zetta | Z |
+|10<sup>-24</sup> | yocto | y  | 10<sup>24</sup> | yotta | Y |
+
+Finally, simple physical units can be combined to complex units. For this, the operators , `*` (multiplication), `/` (division), `**` (power) and `()` (parenthesis) can be used. An example could be
+```
+mV*mV*nS**2/(mS*pA)
+```
+
+Units of the form `<unit> ** -1` can also be expressed as `1\<unit>`. For example
+```
+(ms*mV)**-1
+```
+is equivalent to
+```
+1/(ms*mV)
+```
+
+#### Physical unit literals
+
+Simple unit literals are composed of a number and a type name (with or without a whitespace inbetween the two). Complex unit literals are composed of a number and a type inside of square brackets. Simple units can also be put inside square brackets.
+```
+<number> <unit_type>
+<number> [<unit_type>]
+
+a mV = 1mV
+aa mV = 1[mV]
+b mV/ms = 1[mV/ms]
+```
+
```

Comment:
@DimitriPlotnikov: I don't understand the distinction between simple and complex literals from the point of the user. Do we really need this distinction (internally) and do we have to distinguish between them in the frontend? Also, I think we should either omit the term `literal` here or explain it a bit more. This is computer science terminology which many of our customers probably won't understand.